Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp

```diff
@@ -83,6 +83,29 @@ struct DMABufFeedback {
         {
         }
 
+        explicit FormatTable() = default;
+        explicit FormatTable(uint32_t size, int fd)
+            : size(size)
+            , data(static_cast<FormatTable::Data*>(mmap(nullptr, size, PROT_READ, MAP_PRIVATE, fd, 0)))
+        {
+            if (data == MAP_FAILED) {
+                data = nullptr;
+                size = 0;
+            }
```

Comment:
This is problematic, because we assume that a DMABufFeedback can't be created with an invalid format table (DMABufFeedback::create() fails if mmap fails). For example in `DMABufFeedback::format()` we use the format table without checking it's valid.

Reply:
I understand that the compositor must at least once give a valid `format_table` event during the lifetime of the listener, but indeed it won't hurt to have some check in place before trying to use invalid data.

Another question: should we try to fall-back to read the table from the file descriptor if `mmap()` fails? I suppose it's reasonable to expect that the file descriptor mapping will work given that we should be always receive a valid file descriptor in the event callback... Or does anyone feel differently about this?

Reply:
I'm not sure it's worth it, why would mmap fail but read don't?

Reply:
Let's assume for now that reading the file descriptor would fail also if `mmap()` failed. I tend to think it's probably not worth it to handle, becase mapping would fail if we're running out of virtual address space, and at that point we would have bigger problems.

Reply:
I've added bounds checking in the `format()` method, including a debug assertion because compositors should never reference an invalid format index from a tranche, and they have to provide a format table. There's some graceful handling for release builds.